src/main/java/org/rumbledb/expressions/control/TypeswitchCase.java

```diff
@@ -0,0 +1,59 @@
+package org.rumbledb.expressions.control;
+
+
+import org.rumbledb.expressions.Expression;
+import org.rumbledb.expressions.flowr.FlworVarSequenceType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is a helper class that organizes the children expressions of a typeswitch expression.
+ * From a tree perspective, all expressions in there are considered
+ * to be direct children of the TypeswitchExpression.
+ */
+public class TypeswitchCase {
+
+    private String variableName;
+    private List<FlworVarSequenceType> union = new ArrayList<>();
```

Comment:
How about we moved this initialization to the 1st constructor(with 3 parameters) and explicitly set the union field to null in the 2nd constructor(with 2 parameters, used for the default case). I believe this would both avoid the redundant initialization for the default case as well as potentially making the intention of the 2nd constructor clearer.

Reply:
Very good remark. This is done.